plugins/csv_dataset.cc

```diff
@@ -897,7 +908,7 @@ struct CsvDataset::Itl: public TabularDataStore {
                 //threadAccum.emplace_back(std::move(lineEntry));
             };
 
-        forEachLineBlock(stream, onLine, lineOffset, config.limit);
```

Comment:
If we don't pass lineOffset, how do we a) print the correct line numbers in error messages, and b) deal with the fact that the header is actually a line?  In other words, the for_each_line function was designed to care about *lines* only, and so we need to tell it how many lines we have already skipped when we start so it can tell us the appropriate error message.